src/main/java/com/epam/training/toto/service/TotoService.java

```diff
@@ -1,4 +1,142 @@
+    }
+
+    private Integer getTheLargestPrize() {
+        List<Integer> allPrizes = data.values()
+                .stream()
+                .map(round -> round.getHits())
+                .flatMap(Collection::stream)
+                .map(hit -> hit.getPrizeForTheGames())
+                .collect(Collectors.toList());
+
+        return Collections.max(allPrizes);
+    }
+
+    private List<Map<Outcome, Long>> getDistributionOfResultsForEachRound() {
+        List<Map<Outcome, Long>> distributionForEachRound = new ArrayList<>(2 * data.size());
+        data.values()
+                .forEach(round -> distributionForEachRound.add(round.getOutcomes()
+                        .stream()
+                        .collect(Collectors.groupingBy(Function.identity(), Collectors.counting())))
+                );
+        return distributionForEachRound;
+    }
+
+    public void printTheLargestPrize() {
+        System.out.println("======== The largest prize: " +
+                numberFormat.format(getTheLargestPrize()) +
+                " ========");
+    }
+
+    public void printDistributionOfTheResults() {
+        System.out.println("\n======== The distribution of the results of each round ========");
+        getDistributionOfResultsForEachRound()
+                .forEach(round -> System.out.println("team #1 won: " +
+                        formatDistribution(round.get(Outcome.FIRST_TEAM_VICTORY)) +
+                        ", team #2 won: " +
+                        formatDistribution(round.get(Outcome.SECOND_TEAM_VICTORY)) +
+                        ", draw: " +
+                        formatDistribution(round.get(Outcome.DRAW)))
+                );
+    }
+
+    private String formatDistribution(Long amountOfVictories) {
+        if (amountOfVictories == null) {
+            return "0%";
+        }
+        return String.format("%.2f ", amountOfVictories / (double) NUMBER_OF_GAMES) + "%";
+    }
+
+    public void printHitsAndAmountForTheWager() {
+        System.out.println("\n======== The hits and amount for the wager ========");
+        Optional<Round> round;
+        do {
+            LocalDate dateToSearch = readDateToSearch();
+            round = getRoundByDate(dateToSearch);
+        } while (!round.isPresent());
+
+        List<Outcome> wager = readWager();
+        int hitCounter = 0;
+        int prize = 0;
+        for (int gameCounter = 0; gameCounter < NUMBER_OF_GAMES; gameCounter++) {
+            if (round.get().getOutcomes().get(gameCounter) == wager.get(gameCounter)) {
+                hitCounter++;
+            }
+        }
+        if (hitCounter > 9) {
+            prize = round.get().getHits().get(NUMBER_OF_GAMES % hitCounter).getPrizeForTheGames();
+        }
+        System.out.println("Result: hits: " + hitCounter + ", amount: " + numberFormat.format(prize));
+    }
+
+    private List<Outcome> readWager() {
+        List<Outcome> outcomes = new ArrayList<>();
+        Scanner scanner = new Scanner(System.in);
```

Comment:
scanner is never closed.

Reply:
done

Reply:
closing of one scanner disables another one. So now it's a parameter in TotoService constructor, which is closed in the App after all required methods are called.